design/0003-byop.md

```diff
@@ -0,0 +1,127 @@
+- Bring your own Prometheus
+- @pothulapati
+- Start Date: 2020-04-15
+- Target Date: 
+- RFC PR: [linkerd/rfc#0000](https://github.com/linkerd/rfc/pull/0000)
+- Linkerd Issue: [linkerd/linkerd2#3590](https://github.com/linkerd/linkerd2/issues/3590)
+- Reviewers: @grampelberg @alpeb @siggy
+
+# Summary
+
+[summary]: #summary
+
+This RFC aims to move prometheus as an add-on i.e not mandated
+but adviced. This includes changes to the linkerd-prometheus manifests to make it as a add-on and allowing more configuration (more on this later),
+along with documentation and tooling to use existing prometheus installation.
+
+# Problem Statement
+
+[problem-statement]: #problem-statement
+
+Lots of users already have a exisiting prometheus installation in their cluster and don't like managing
+another prometheus instance especially because they could be resource intensive and painful. They usually
+have a prometheus installation paired up with a long term monitroing solutions like Cortex,InfluxDB, or third party
+solutions like Grafana Cloud, etc.
+In those cases prometheus is just a temporary component, that keeps pushing metrics into their long term solution, making users want to re-use their exisitng prometheus.
+
+The following are the list of problems:
+
+- Hard requirement of linkerd-prometheus.
+
+- Configuring the current linkerd-prometheus for remote-writes, etc is hard, because the cm is overriden after upgrades, etc.
+
+- Not enough documentation for bringing your own prometheus case.
+
+Once the RFC is implemented, Apart from the default prometheus enabled installation Users should be
+able to use their exising prometheus instance, to also scrape metrics from the proxy
+
+# Design proposal
+
+[design-proposal]: #design-proposal
+
+## Prometheus as an Add-On
```

Comment:
I would really like to see a discussion about how we're planning on using their prometheus from the linkerd codebase, what could go wrong and how we can make that connection less fragile.